sw/device/tests/alert_handler_reverse_ping_in_deep_sleep_test.c

```diff
@@ -0,0 +1,229 @@
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = ARRAYSIZE(alerts),
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = alert_handler_testutils_get_cycles_from_us(
+          kTestParamAlertHandlerPingTimeoutUsec),
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+}
+
+/**
+ * External ISR.
+ *
+ * Handles all peripheral interrupts on Ibex. PLIC asserts an external interrupt
+ * line to the CPU, which results in a call to this OTTF ISR. This ISR
+ * overrides the default OTTF implementation.
+ */
+void ottf_external_isr(void) {
+  top_earlgrey_plic_peripheral_t peripheral_serviced;
+  dif_alert_handler_irq_t irq_serviced;
+  isr_testutils_alert_handler_isr(plic_ctx, alert_handler_ctx,
+                                  &peripheral_serviced, &irq_serviced);
+  CHECK(peripheral_serviced == kTopEarlgreyPlicPeripheralAlertHandler,
+        "Interurpt from unexpected peripheral: %d", peripheral_serviced);
+  interrupt_serviced = true;
+}
+
+bool test_main(void) {
+  init_peripherals();
+
+  if (pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) {
+    LOG_INFO("POR reset");
+    CHECK(rstmgr_testutils_reset_info_any(&rstmgr, kDifRstmgrResetInfoPor));
+    rstmgr_testutils_pre_reset(&rstmgr);
+
+    alert_handler_config();
+
+    irq_global_ctrl(true);
+    irq_external_ctrl(true);
+
+    uint32_t wakeup_threshold = aon_timer_testutils_get_aon_cycles_from_us(
+        kTestParamWakeupThresholdUsec);
+
+    // Sleep longer in FPGA and silicon targets.
+    if (kDeviceType != kDeviceSimDV && kDeviceType != kDeviceSimVerilator) {
+      uint32_t wakeup_threshold_new = wakeup_threshold * 100;
+      CHECK(wakeup_threshold_new > wakeup_threshold,
+            "Detected wakeup_threshold overflow.");
+      wakeup_threshold = wakeup_threshold_new;
+    }
+
+    // Enable and enter deep sleep.
+    aon_timer_testutils_wakeup_config(&aon_timer, wakeup_threshold);
+    pwrmgr_testutils_enable_low_power(&pwrmgr,
+                                      kDifPwrmgrWakeupRequestSourceFive, 0);
+    wait_for_interrupt();
+    CHECK(false, "Fail to enter in low power mode!");
+    OT_UNREACHABLE();
+  } else if (pwrmgr_testutils_is_wakeup_reason(
+                 &pwrmgr, kDifPwrmgrWakeupRequestSourceFive)) {
+    LOG_INFO("Wakeup reset");
```

Comment:
Will this generate a RESET? If so, I wonder if it will clear alert_handler's cause register? I think it would still be good to check the alert_cause and loc_alert_cause, but is there any information that will maintain during reset? Maybe `rst_cause`?